aviary/subsystems/propulsion/utils.py

```diff
@@ -135,13 +136,45 @@ def build_engine_deck(aviary_options: AviaryValues):
             try:
                 # add value from aviary_options to engine_options
                 aviary_val = aviary_options.get_val(var, units)
+                default_value = meta_data[var]['default_value']
+                # special handling for iterables - check if they are multidimensional,
+                # which implies multiple engine models, and only use the value intended
+                # for the first engine model
                 if isinstance(aviary_val, np.ndarray):
+                    try:
+                        expected_dim = default_value.ndim
+                    except AttributeError:
+                        # metadata is not expecting a numpy array, just a single value
+                        expected_dim = 0
```

Comment:
would a getattr with a default value be better here?

Reply:
This is just checking if we are expecting an array by checking metadata, and if so how many dimensions that array should be. Never actually overwriting provided values with a default

Reply:
FYI I just overhauled the try/except statement structure so this code is about to become outdated